WordPress/gutenberg
# Site Editor Navigation Commands: Add permission check

Fixes #62460

## What?
This PR prevents users without `edit_theme_options` permissions from unintentionally seeing commands related to accessing the site editor.

The commands that are displayed unintentionally are Page, Template, and Template Part. Users without the `edit_theme_options` capability don't have access to the site editor, so they will get a warning when running these commands:

```
Sorry, you are not allowed to access this page.
```

## Why?

The current command palette checks whether the site editor is accessible via code like the following:

```javascript
const canCreateTemplate = useSelect( ( select ) => {
	return select( coreStore ).canUser( 'create', {
		kind: 'postType',
		name: 'wp_template',
	} );
}, [] );
```

However, this check did not exist for the three commands mentioned above.

### How?

Check whether the template can be created, i.e. whether the site editor is accessible.

If the site editor is not accessible, I either link to the post editor or don't display the command at all.

Whether a template can be created and whether the site editor is accessible may not necessarily coincide in the future, but I believe this is the best approach for now.

## Testing Instructions

- Open the post editor.
- Test the following four scenarios to confirm that commands appear or do not appear correctly, and that navigation works correctly.

Note: The scenarios marked with ✅ mark are the ones that will be fixed by this PR.

- Classic Theme, Admin user / Editor user:
  - `Hello World` (post) > Post Editor
  - `Sample page` (page) > Post Editor
- Block Theme, Admin user:
  - `Hello World` (post) > Post Editor
  - `Sample page` (page) > Site Editor
  - `Blog Home` (template) > Site Editor
  - `Post Meta` (template part) > Site Editor
- Block Theme, Editor user:
  - `Hello World` (post) > Post Editor
  - ✅ `Sample page` (page) > Post Editor
  - ✅ `Blog Home` (template) > Do not show
  - ✅ `Post Meta` (template part) > Do not show
